src/apps/fuzzers/fuzzerPolygonToCellsNoHoles.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @brief Fuzzer program for polygonToCells and related functions, without holes
+ */
+
+#include "aflHarness.h"
+#include "h3api.h"
+#include "utility.h"
+
+const int MAX_RES = 15;
+const int MAX_SZ = 4000000;
+
+void run(GeoPolygon *geoPolygon, int res) {
+    int64_t sz;
+    H3Error err = H3_EXPORT(maxPolygonToCellsSize)(geoPolygon, res, &sz);
+    if (!err && sz < MAX_SZ) {
+        if (sz < 0) {
+            // TODO: Check on this once rebased
+            printf("Oh no - sz is negative\n");
+        }
+        H3Index *out = calloc(sz, sizeof(H3Index));
+        H3_EXPORT(polygonToCells)(geoPolygon, res, out);
+        free(out);
+    }
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    if (size < sizeof(int)) {
+        return 0;
+    }
+
+    uint8_t res = *data;
+    size_t vertsSize = size - 1;
+    int numVerts = vertsSize / sizeof(LatLng);
+
+    GeoPolygon geoPolygon;
+    geoPolygon.numHoles = 0;
+    geoPolygon.holes = NULL;
+    geoPolygon.geoloop.numVerts = numVerts;
+    geoPolygon.geoloop.verts = (LatLng *)(data + 1);
```

Comment:
For my info - why `+ 1`? Oh, because the first `uint_8` is the res? Might be worth a comment.

I imagine that most fuzzer runs would error quickly on an invalid `res` here, and never exercise the polyfill...

Reply:
This is still better than fuzzerPoylgonToCells.c which is very difficult for the fuzzer to make progress on; here it only needs to guess one byte correctly (polygonToCells actually takes an int but I don't think that's quite as relevant) which I believe is very doable.